cmd/prometheus/main.go

```diff
@@ -544,20 +550,20 @@ func main() {
 
 	// Throw error for invalid config before starting other components.
 	var cfgFile *config.Config
-	if cfgFile, err = config.LoadFile(cfg.configFile, agentMode, false, log.NewNopLogger()); err != nil {
+	if cfgFile, err = config.LoadFile(cfg.configFile, agentMode, false, slog.New(slog.NewTextHandler(io.Discard, nil))); err != nil {
```

Comment:
Suggest creating a utility function for `slog.New(slog.NewTextHandler(io.Discard, nil))` which occurs 147 times in this PR.

Reply:
Sure, that's reasonable. 

Would you rather that live here in Prometheus proper or perhaps as a utility function in prometheus/common (`promslog.NewNopLogger`)?

Reply:
No strong opinions. If there is a package that a large part of the 147 cases are already importing I’d look at using that, but don’t make it weird.

Reply:
https://github.com/prometheus/common/pull/697

feedback welcome!